libraries/AP_AdvancedFailsafe/AP_AdvancedFailsafe.h

```diff
@@ -64,19 +64,28 @@ class AP_AdvancedFailsafe
         }
 
     // check that everything is OK
-    void check(enum control_mode control_mode, uint32_t last_heartbeat_ms, bool geofence_breached, uint32_t last_valid_rc_ms);
+    void check(uint32_t last_heartbeat_ms, bool geofence_breached, uint32_t last_valid_rc_ms);
 
     // generate heartbeat msgs, so external failsafe boards are happy
     // during sensor calibration
     void heartbeat(void);
 
-    // called in servo output code to set servos to crash position if needed
-    void check_crash_plane(void);
-
+    // return true if we are terminating (deliberately crashing the vehicle)
+    bool should_crash_vehicle(void);
+    
     // for holding parameters
     static const struct AP_Param::GroupInfo var_info[];
 
-private:
+    // called to set all outputs to termination state
+    virtual void terminate_vehicle(void) = 0;
+        
+protected:
+    // setup failsafe values for if FMU firmware stops running
+    virtual void setup_IO_failsafe(void) = 0;
```

Comment:
I've been treating this as a per-class decision. Personally I find the underscores make the code significantly harder to read, and I have not been using them for new classes that I write for either methods or variables.